src/components/Checkbox/Checkbox.vue

```diff
@@ -1,50 +1,50 @@
 <template>
-  <div  :class='[cursorClass, disabledClass]'>
-  <SwitchGroup  :disabled='disabled'>
-    <div class='flex items-center'>
-      <SwitchLabel :class='[`mr-4`, cursorClass]'>{{ srLabel }}</SwitchLabel>
+  <component :is="formElement ? CFormElement : CFragment" v-bind="{ id, label, size, color, labelFormatter, message, stacked, expandInput: false, }">
+    <div class="concrete__checkbox flex">
+      <span class="sr-only">{{srLabel}}</span>
       <Switch 
         :disabled='disabled' 
         :id='id' 
         v-model='enabled' 
-        :class='[`absolute right-4 items-center border-2`, `h-${sized} w-${sized}`, cursorClass]'>
-          <CheckIcon v-if='modelValue' class='object-scale-down justify-center' />
+        class="items-center border border-gray-300"
+        :class='[`h-${sized} w-${sized}`, cursorClass, bgColor]'>
+        <span class="flex items-center justify-center transition-opacity" >
+          <slot v-if="modelValue === true">
+            <CheckIcon
+              :class="[``, checkIconClass]"
+            />
```

Comment:
minor formatting - this can just be a single line, and I don't understand the empty string in the array. Can this be simplified to:
```suggestion
            <CheckIcon :class="checkIconClass" />
```

Reply:
good spot!